test/sdk.test.ts

```diff
@@ -0,0 +1,130 @@
+import { SDK } from '../index';
+import { it, describe, expect, vi, afterEach } from 'vitest'
+import { importWasms, generateProof } from '../noir';
+
+describe('constructor()', () => {
+  it('should set chainName, chainId and rpcUrl', () => {
+    const sdk = new SDK({ chainName: 'mainnet', rpcUrl: 'https://rpc.url' });
+
+    expect(sdk.chainName).toBe('mainnet');
+    expect(sdk.rpcUrl).toBe('https://rpc.url');
+    expect(sdk.chainId).toBe(1);
+  });
+
+  it('should throw an error if chain is not viem supported', () => {
+    try {
+      new SDK({ chainName: 'invalid', rpcUrl: 'https://rpc.url' });
+    } catch (error) {
+      expect((error as Error).message).toBe('Chain invalid is not supported');
+    }
+  });  
+
+  it('should throw an error if chain is not vlayer supported', () => {
+    try {
+      new SDK({ chainName: 'polygon', rpcUrl: 'https://rpc.url' });
+    } catch (error) {
+      expect((error as Error).message).toBe('Chain polygon is not supported');
```

Comment:
This seems like a word order issue. Let's do either `Polygon chain is not supported` or `Polygon is not supported`. Or `Unsupported chain: Polygon`, so that we can later lookup by prefix